gui/src/components/HomeComponents/CreateOrEditProjectBody.tsx

```diff
@@ -48,6 +85,37 @@ export default function CreateOrEditProjectBody({
   const projectIdRef = useRef(null);
   const router = useRouter();
 
+  const [integrationLoading, setIntegrationLoading] = useState<boolean>(false);
+  const [isExternalGitIntegration, setIsExternalGitIntegration] =
+    useState<boolean>(false);
+  const [useExternalGit, setUseExternalGit] = useState<boolean>(false);
+  const [repositories, setRepositories] = useState<any[]>([]);
+  const [selectedRepository, setSelectedRepository] = useState<{
+    label: string;
+    value: string;
+  } | null>(null);
+
+  async function redirectToGithubIntegration() {
+    setIntegrationLoading(true);
+    try {
+      const interval = setInterval(
+        async () => {
+          const gitIntegrated = await isGithubConnected();
+          if (gitIntegrated) {
+            setIsExternalGitIntegration(true);
+            setIntegrationLoading(false);
+            clearInterval(interval);
+          }
+        },
+        1000,
```

Comment:
setInterval function normally takes only two primary arguments, the callback and the delay, so Ig the second 1000 is redundant.